### PR TITLE
SDK release pipeline

### PR DESCRIPTION
**Pull request by AlbertoBelver**
[AB#126924](https://dev.azure.com/niposoftware/Nfield/_sprints/taskboard/Team%20Yellow/Nfield/Release%202023/2023%20P2S6?workitem=126924)

Related PRs:
https://github.com/NIPOSoftwareBV/Nfield-Documentation/pull/3506

### CODE REVIEW

File: Pipelines/YmlTemplates/postBuildSteps.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Shared post build template, run after the build template. Currently used by
+# - Domain API pipeline
```
Comment (tdraganidis) [inline Pipelines/YmlTemplates/postBuildSteps.yml:2]: Is this a copy-paste issue or it's correctly mentioning Domain API pipeline?

File: Pipelines/YmlTemplates/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                      Write-Host "##vso[task.setvariable variable=BranchName]$BranchName"
+                      Write-Host "##vso[task.setvariable variable=ReleaseName]$ReleaseName"
+                      Write-Host "##vso[task.setvariable variable=CommitHash]$CommitHash"
+                      Write-Host "##vso[task.setvariable variable=VersionName]$VersionName"
+                      
+                - task: GitHubRelease@1
+                  displayName: 'Create GitHub Release'
+                  condition: ${{ eq(parameters.nuGetFeedType, 'external') }}
+                  inputs:
+                    gitHubConnection: 'github.com_AlbertoBelver' # OAuth Service Conection https://dev.azure.com/niposoftware/Nfield/_settings/adminservices
```
Comment (tdraganidis) [inline Pipelines/YmlTemplates/deploy.yml:74]: Shouldn't we have a proper github connection for that? And not a personal one?

File: Pipelines/Scripts/GetNugetVersion.ps1 [hunk 3]
```diff
@@ -34,20 +34,38 @@ if(([regex]::Matches($VersionFormat, "\." )).count -ne 2)
...
 {
-    $Branch = $env:BUILD_SOURCEBRANCHNAME
-    Write-Host "Branch:" $Branch
-    $Suffix = if ($Branch -eq "master") {"-beta"} else {"-alpha"}	
+    Write-Host "Branch:" $branchName
+    $Suffix = if ($branchName -like 'master') {'-beta'} else {'-alpha'}	
     Write-Host "Suffix:" $Suffix
 }
 
+$releaseName = $branchName.Split("/");
```
Comment (tdraganidis) [inline Pipelines/Scripts/GetNugetVersion.ps1:52]: You may give an example of what you are expecting here in order to make it easier to understand (like release/2023-P3S5). 
In the else clause, do you expect it to be equal to 0, 1 or more than 2? Can you give an example of what is expected there?

File: README.md [hunk 4]
```diff
@@ -72,3 +72,13 @@ For feedback related to this SDK please visit the
 [Nfield website].
 
 [Nfield website]: https://www.nipo.com/
+
+## Releases
+ - How to create and publish a new `-alpha` or `-beta` version. `-alpha` and `-beta` versions can only be published on a DevOps private/internal Nfield Feed:
+    - Create a new branch with the this patern : `*/ci-*` or merge a commit into `master`
```
Comment (tdraganidis) [inline README.md:78]: `nitpick` Typo: `with this pattern`

File: README.md [hunk 5]
```diff
@@ -72,3 +72,13 @@ For feedback related to this SDK please visit the
...
+    - This branch will trigger a new build in DevOps pipelines
+    - After the build a `Release pipeline` will allow to Push the new nuget version in the `Nfield` internal feed. These versions __don't create__ `GitHub releases`.
+ - How to create and publish a new `release` on the Nuget.org public Feed
+    - Create a new release branch Example: `release/2023-P3S5`
+    - This branch will trigger a new build in DevOps pipelines
+    - After the build a `Release pipeline` will allow to Push the new nuget version in the `Nuget.org` feed and it will create the `GitHub release` repo.
```
Comment (tdraganidis) [inline README.md:84]: Are there any approvals in this process?

File: Pipelines/deploy-nuget.yml [hunk 6]
```diff
@@ -1,17 +1,51 @@
...
+    steps:
+    - task: ManualValidation@0
+      timeoutInMinutes: 1440 # task times out in 1 day
+      inputs:
+        instructions: 'Please validate the the nuget publishing'
+        onTimeout: 'reject'
 
-variables:
-  buildConfiguration: 'Release'
+####################### INTERNAL DEPLOY NFILED ####################
```
Comment (tdraganidis) [inline Pipelines/deploy-nuget.yml:38]: `nitpick` Typo: Nfield instead of Nfiled

**Review by tdraganidis — COMMENTED** [review]
Nice job @AlbertoBelver ! Some nitpicking mainly and couple of comments. 
- I would also value some documentation on the beginning of each yml template. I think that we have an agreement on adding a description and some parameters explanation there.

File: Pipelines/YmlTemplates/postBuildSteps.yml (hunk 1 above)
Comment (AlbertoBelver) [reply Pipelines/YmlTemplates/postBuildSteps.yml:2]: copy-paste

File: Pipelines/YmlTemplates/deploy.yml (hunk 2 above)
Comment (AlbertoBelver) [reply Pipelines/YmlTemplates/deploy.yml:74]: We don't have any shared github account to do it. I want to discuss it in the SoS meeting. By the moment I leave it as it is.
Is also happening with Github PATs

Comment (AlbertoBelver) [reply Pipelines/YmlTemplates/deploy.yml:74]: https://github.com/NIPOSoftwareBV/Nfield-Documentation/blob/master/meeting-notes/Retros/SoS-discussion-points.md

File: README.md (hunk 5 above)
Comment (AlbertoBelver) [reply README.md:84]: Yes is using an [Environment](https://dev.azure.com/niposoftware/Nfield/_environments/104?view=deployments) that only OPS can approve

**Review by tdraganidis — APPROVED** [review]

**Pull request merged**